cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -216,6 +216,20 @@ void CheckRoundTripResult(const std::shared_ptr<Table> expected_table,
   compute::AssertTablesEqualIgnoringOrder(merged_expected, output_table);
 }
 
+void countProjectNodeOptionsInDeclarations(const compute::Declaration& input,
+                                           int& counter) {
```

Comment:
Can you change this to just return an `int` instead of taking an out parameter?

Reply:
I added a static counter, but I am not confident with this change. Could you please check?